Source/WebCore/loader/CookieJar.cpp

```diff
@@ -63,6 +64,26 @@ IncludeSecureCookies CookieJar::shouldIncludeSecureCookies(const Document& docum
     return (url.protocolIs("https"_s) && !document.foundMixedContent().contains(SecurityContext::MixedContentType::Active)) ? IncludeSecureCookies::Yes : IncludeSecureCookies::No;
 }
 
+SameSiteInfo CookieJar::sameSiteInfo(const Document& document, const URL& url, IsForDOMCookieAccess isAccessForDOM)
```

Comment:
I don't think we need the URL parameter? You can probably compare the document.url() and document.firstPartyForCookies() to determine if this service worker is for a first party.

See `Page::setupForRemoteWorker()` for how the Document is set up for remote workers.